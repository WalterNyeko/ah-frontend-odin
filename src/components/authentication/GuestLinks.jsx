import React from 'react';
import PropTypes from 'prop-types';

const GuestLinks = ({ showModal }) => (
  <React.Fragment>
    <li className="nav-item">
      <button
        className="nav-link text-light btn-link"
        onClick={() => showModal('login')}
        data-id="login"
      >
        login
      </button>
    </li>
    <li className="nav-item">
      <button
        className="nav-link text-light btn-link"
        onClick={() => showModal('signup')}
        data-id="signup"
      >
        Register
      </button>
    </li>
  </React.Fragment>
);

GuestLinks.propTypes = {
  showModal: PropTypes.func.isRequired,
};

export default GuestLinks;
