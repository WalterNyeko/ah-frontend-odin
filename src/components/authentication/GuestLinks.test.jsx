import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GuestLinks from './GuestLinks';

describe('GuestLinks', () => {
  test('It renders the same UI', () => {
    expect(renderer.create(<GuestLinks showModal={jest.fn()} />).toJSON()).toMatchSnapshot();
  });

  test('login button triggers show login modal', () => {
    const showModal = jest.fn();
    const wrapper = shallow(<GuestLinks showModal={showModal} />);

    wrapper.find('button[data-id="login"]').simulate('click');
    expect(showModal).toHaveBeenCalledWith('login');
  });

  test('signup button triggers show signup modal', () => {
    const showModal = jest.fn();
    const wrapper = shallow(<GuestLinks showModal={showModal} />);

    wrapper.find('button[data-id="signup"]').simulate('click');
    expect(showModal).toHaveBeenCalledWith('signup');
  });
});
