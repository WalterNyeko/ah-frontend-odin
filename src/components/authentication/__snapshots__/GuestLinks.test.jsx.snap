// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestLinks It renders the same UI 1`] = `
Array [
  <li
    className="nav-item"
  >
    <button
      className="nav-link text-light btn-link"
      data-id="login"
      onClick={[Function]}
    >
      login
    </button>
  </li>,
  <li
    className="nav-item"
  >
    <button
      className="nav-link text-light btn-link"
      data-id="signup"
      onClick={[Function]}
    >
      Register
    </button>
  </li>,
]
`;
